### PR TITLE
json type definition templates / timeline item type filter

### PR DESCRIPTION
**Pull request by nk-coding**
- also adds a search query for IssuePriority
- fixes templated field nullability behavior

### CODE REVIEW

File: core/src/main/kotlin/gropius/util/schema/Schema.kt [hunk 1]
```diff
@@ -0,0 +1,183 @@
...
+        verify(this)
+    }
+
+    /**
+     * Verifies that this schema is valid
+     *
+     * @param root the root schema
+     * @throws InvalidSchemaException if the schema is not valid
+     */
+    private fun verify(root: Schema) {
```
Comment (chriku) [inline core/src/main/kotlin/gropius/util/schema/Schema.kt:45]: unnecessary long, please butcher

File: core/src/main/kotlin/gropius/util/schema/Validator.kt [hunk 2]
```diff
@@ -0,0 +1,338 @@
...
+     * @param instance the JSON data to validate
+     * @param state the validation state, used to track errors
+     * @param parentTag the parent tag, used to prevent infinite recursion
+     */
+    private fun validate(state: ValidationState, schema: Schema, instance: JsonNode, parentTag: String?) {
+        if (schema.nullable && instance.isNull) {
+            return
+        }
+
+        when (schema.form) {
```
Comment (chriku) [inline core/src/main/kotlin/gropius/util/schema/Validator.kt:71]: maybe functionate out every choice?

**Review by chriku — CHANGES_REQUESTED** [review]
consider using two PRs for two different and (seemingly) easy splittable features

File: core/src/main/kotlin/gropius/util/schema/Schema.kt (hunk 1 above)
Comment (nk-coding) [reply core/src/main/kotlin/gropius/util/schema/Schema.kt:45]: this is a fork of the original json type definition validator
I would prefer leave this code as close as possible to the original to make reacting to changes in the original simpler, in particular in the validation code

File: core/src/main/kotlin/gropius/util/schema/Validator.kt (hunk 2 above)
Comment (nk-coding) [reply core/src/main/kotlin/gropius/util/schema/Validator.kt:71]: see https://github.com/ccims/gropius-backend/pull/72#discussion_r1450819139

**Review by chriku — APPROVED** [review]

**Pull request merged**